src/platform/qt/ScriptingController.cpp

```diff
@@ -7,11 +7,12 @@
 
 #include "CoreController.h"
 #include "ScriptingTextBuffer.h"
+class QTextDocument;
 
 using namespace QGBA;
 
 ScriptingController::ScriptingController(QObject* parent)
-	: QObject(parent)
+	: QAbstractListModel(parent)
```

Comment:
I'm...really not sure how I feel about making the ScriptingController inherit from QAbstractListModel. But at the same time, splitting out a separate class also feels...strange.

Reply:
The same thought occurred to me while I was writing it. It felt a little bit strange from a conceptual level, but the alternative is a thin little wrapper class that mostly just proxies signals, which seemed to be a bit much. I'll leave it up to you if that's something you want to change.

Reply:
I thought about this more, and I really don't like it. The model represents something arbitrarily associated with scripting, the text buffers, not scripts themselves. The moment something else comes up that might want to be stuck in a list, this won't work. For forward-thinking purposes, I think it needs to be refactored into a separate class.

Reply:
That's entirely fair, and it does satisfy the uncertainty I had. Done.